CommunityFridgeMapApi/dependencies/python/s3_service.py

```diff
@@ -0,0 +1,101 @@
+import os
+import uuid
+from urllib.parse import urlparse, urlunparse
+import boto3
+import botocore
+from botocore.config import Config
+
+def get_s3_client(env=os.getenv("Environment")):
+    endpoint_url="http://localstack:4566/" if env == "local" else None
+    config = Config(
+        signature_version=botocore.UNSIGNED, # Do not include signatures in s3 presigned-urls.
+    )
+
+    return boto3.client(
+        "s3",
+        config=config,
+        endpoint_url=endpoint_url,
+    )
+
+def translate_s3_url_for_client(url: str, env=os.getenv("Environment")) -> str:
+    """
+    This function translates a S3 url to a client-accessible urls.
+
+    From a lambda function's perspective, local AWS gateway is located at "localstack:4566" (within cfm-network).
+    However, this hostname is not available for the host machine.
+    In order to fetch S3 files from a local browser, we need to use "localhost:4566".
+    """
+    if env == "local":
+        parsed_url = urlparse(url)
+        return urlunparse(parsed_url._replace(netloc="localhost:4566"))
+    return url
+
+
+class S3ServiceException(Exception):
+    pass
+
+
+class S3Service:
+    """
+    Adapter class for persisting binary files in S3 buckets.
+    """
+    def __init__(self):
+        self._client = get_s3_client()
+
+    def idempotent_create_bucket(self, bucket: str):
+        """
+        creates a bucket if there is no existing bucket with the same name.
+            Parameters:
+                bucket: name of the bucket
+        """
+        try:
+            self._client.create_bucket(Bucket=bucket)
+        except:
```

Comment:
It's dangerous catching all the Exceptions. It makes it hard to debug and can result in errors/bugs going untreated. I think it would be better if we caught specific exceptions. 

I got this error when uploading an image in the deployed lambda: `{"error": "Unexpected error prevented server from fulfilling request."}`
    * `curl --request POST   --url https://0bebq1ckx7.execute-api.us-east-1.amazonaws.com/Prod/v1/image   --header 'Content-Type: image/webp'   --data-binary "@/Users/jtorre-admin/Downloads/DriversLicense.webp"`

However `curl --request POST   --url https://0bebq1ckx7.execute-api.us-east-1.amazonaws.com/Prod/v1/image   --header 'Content-Type: image/webp'   --data-binary "@/Users/jtorre-admin/Downloads/DriversLicense.jpeg"` returned the expected response
    * `{"error": "Request could not be understood due to incorrect syntax."}`

Reply:
Or when it's a more ambiguous exception. We should log the exception so we can debug like this: https://github.com/CollectiveFocus/CFM_Backend/blob/dev/CommunityFridgeMapApi/dependencies/python/db.py#L115

Reply:
Yes, it does sound like `template.yaml` misconfiguration. Do we need to specify s3 buckets in that file?
Added some logging here: 66e70f19a9453f874a179ea5559eff0a418e687e